src/Control/Effect/Cull.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE DeriveFunctor, ExistentialQuantification, FlexibleContexts, FlexibleInstances, LambdaCase, MultiParamTypeClasses, StandaloneDeriving, TypeOperators, UndecidableInstances #-}
+module Control.Effect.Cull where
```

Comment:
I'm on the fence on this one too. The `once` terminology is more established, but I always liked the play on words that `cull` has to offer.

Looking further afield, `call`, `once`, `list` (or `all`) are all specific versions of something that handles nondeterminism in a particular way. The heuristics entwined paper offers lots of different interpretations. My point here is that we ought to decide what to name the generic operation, and then these interpretations are just algebras over some scoped syntax. 

As for breaking changes, I hope we're not locked in to decisions about terminology already! I'd quite like it if we spent some time consolidating names so that they're not too far from what's in the literature.

Reply:
> As for breaking changes, I hope we're not locked in to decisions about terminology already! I'd quite like it if we spent some time consolidating names so that they're not too far from what's in the literature.

Yeah, that’s a fair point. I think we can make breaking changes as long as we bump the version appropriately (I’ve been informally following semantic versioning rules).

Reply:
I personally think `cull` is the most immediately-meaningful terminology, and I agree with @zenzike that a play-on-words term is good and memorable. I’d be in favor of implementing `runNonDetOnce` in terms of `cull`. It does indeed have a low Levenshtein distance from `call`, but given English's glut of four-letter words, I think that battle is not worth fighting 😄 

> I think we can make breaking changes as long as we bump the version appropriately

I agree. I think there’s an implicit understanding that versions < 1 can have breaking changes.